### PR TITLE
feat(node): support conditional exports

### PR DESCRIPTION
**Pull request by kt3k**
This PR adds the support of the [conditional exports](https://nodejs.org/api/packages.html#packages_conditional_exports) of Node.js module resolution.

The packages like [colorette](https://www.npmjs.com/package/colorette), [postcss](https://www.npmjs.com/package/postcss), etc use conditional exports in their package.json. We need to handle `exports` field of package.json correctly to support these modules.

This PR adds `colorette@1.2.1` as the fixture for testing the conditional exports.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -60,7 +60,7 @@ jobs:
         run: deno upgrade --canary
 
       - name: Format
-        run: deno fmt --check
+        run: deno fmt --check --ignore=node/_module/node_module
 
       - name: Lint
-        run: deno lint --unstable
+        run: deno lint --unstable --ignore=node/_module/node_modules
```
Comment (bartlomieju) [inline .github/workflows/ci.yml:66]: Can you add ignores inline to appropriate files so we can keep these commands without additional arguments?

File: node/_module/node_modules/colorette/README.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
+# Colorette [![](https://img.shields.io/npm/v/colorette.svg?label=&color=2a64e6)](https://www.npmjs.org/package/colorette)
+
+> Color your terminal using pure idiomatic JavaScript.
+
+Colorette is a Node.js library for embellishing your CLI tools with colors and
```
Comment (bartlomieju) [inline node/_module/node_modules/colorette/README.md:5]: This file is probably not needed?

File: node/_module/node_modules/colorette/LICENSE.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
+Copyright © Jorge Bucaran <<https://jorgebucaran.com>>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of
+this software and associated documentation files (the 'Software'), to deal in
+the Software without restriction, including without limitation the rights to
+use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
```
Comment (bartlomieju) [inline node/_module/node_modules/colorette/LICENSE.md:8]: Maybe just add first line to appropriate files and remove this file?

**Comment by kt3k** [comment]
@bartlomieju Thank you for your comments! I updated the PR to include as few fixture files as possible.

**Review by bartlomieju — APPROVED** [review]
LGTM, thanks for simplifying @kt3k

**Pull request merged**